### PR TITLE
E3SM UXarray and Dask Examples 

### LINKED ISSUE #822: E3SM Visualization Example
A notebook showcasing basic plots of E3SM model output to put in our visualization gallery.

### PR DESCRIPTION
**Pull request by rytam2**
Closes #822

## Overview

Added user guide notebook on parallel loading with Dask; and workflow examples with E3SM output with visualization and calculations. 

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Examples**
- [x] Any new notebook examples added to `docs/examples/` folder
- [x] Clear the output of all cells before committing
- [x] New notebook files added to `docs/examples.rst` toctree
- [x] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`

### CODE REVIEW

**Comment by rytam2** [comment]
<p>Hi Philip, thanks for the comment! For the syntax, do you mean I'll do the following: </p>

<pre class="ql-syntax" spellcheck="false">$$
{SWCRE = FSNT-FSNTC}
$$
</pre><p>Or do you mean adding <code>{math}</code> before the <code>SWCRE...FSNTC</code> equation, and have <code>$$</code> removed?  </p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2214665974'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by philipc2** [comment]
<p>Correct, it should look like {math}{SWCRE = FSNT - FSNTC}</p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2214665974'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by philipc2** [comment]
Looks like the LaTex is still broken. Can you add the following to our `conf.py` and see if it fixes things? 

https://github.com/UXARRAY/uxarray/blob/main/docs/conf.py

```Python
myst_enable_extensions = [
    "amsmath",
    "attrs_inline",
    "colon_fence",
    "deflist",
    "dollarmath",
    "fieldlist",
    "html_admonition",
    "html_image",
    "linkify",
    "replacements",
    "smartquotes",
    "strikethrough",
    "substitution",
    "tasklist",
]
```

**Comment by philipc2** [comment]
@rytam2 

Now that you've marked the PR as ready for review, there's a few things that you can also do: 
- Remove the DRAFT from the title 
- Add reviewers (can add orhan, rajeev, aaron and myself) 

image

**Comment by philipc2** [comment]
@anissa111 

Any idea on why LatEx isn't rendering correctly in the notebooks? We've tried a few things above but are still having some issues.

**Comment by anissa111** [comment]
> @anissa111
> 
> Any idea on why LatEx isn't rendering correctly in the notebooks? We've tried a few things above but are still having some issues.

I'll take a look!

**Comment by anissa111** [comment]
Okay, locally either

````
```{math}
	SWCRE = FSNT - FSNTC
```
````

or

```
$$
	SWCRE = FSNT - FSNTC
$$
```

Works for me

**Comment by rytam2** [comment]
<p>Now updated with cluster usage. </p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2232052027'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by rytam2** [comment]
<p>Moved above prior to closing cluster.</p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2232052042'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by erogluorhan — COMMENTED** [review]
Great additions! I have some inline and ReviewNB comments below

File: docs/conf.py [hunk 1]
```diff
@@ -199,7 +199,12 @@ def __getattr__(cls, name):
 autodoc_typehints = "none"
 
 # notebooks to exclude from run-time execution, will use rendered version
-nb_execution_excludepatterns = ["3_75km_mpas.ipynb"]
+nb_execution_excludepatterns = [
+    "examples/visualization/3_75km_mpas.ipynb",
+    "examples/e3sm-calc-workflow.ipynb",
+    "examples/e3sm-load-viz.ipynb",
+    "user-guide/parallel-load-ux-with-dask.ipynb",
+]
```
Comment (philipc2) [inline docs/conf.py:207]: We can change this back to the name of the notebooks, without the paths.

**Comment by rytam2** [comment]
<p>Added paragraph from zulip discussion. </p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2243731037'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by rajeeja** [comment]
We might want to think about changing the name of `Gallery` - that houses two of the notebooks in this PR.

The formatting of equations for the below: don't look good://
Screenshot 2024-08-12 at 13 51 03
//

**Comment by philipc2** [comment]
> We might want to think about changing the name of `Gallery` - that houses two of the notebooks in this PR.
> 
> The formatting of equations for the below: don't look good:// Screenshot 2024-08-12 at 13 51 03 //

The Gallery can house both Workflow & Visualization examples. Do you think this sepperation is appropriate? 

image

The equations look like this for me in the rendered docs in light mode. 

image


Looks like they are broken with the Dark Mode.

image

Have you ever encountered this @anissa111 ?

**Comment by anissa111** [comment]
I haven't actually. 

I poked around for a bit and suggest starting at this [stackoverflow post](https://stackoverflow.com/questions/11096214/color-in-mathjax) and here in [mathjax documentation](https://docs.mathjax.org/en/v2.2-latest/CSS-styles.html).

Let me know what you find!

**Comment by rljacob** [comment]
I think of "gallery" as just being images/viz.

**Review by rajeeja — CHANGES_REQUESTED** [review]
see comments in e3sm-load notebook.

File: docs/userguide.rst [hunk 2]
```diff
@@ -31,8 +31,11 @@ These user guides provide detailed explanations of the core functionality in UXa
 `Data Structures <user-guide/data-structures.ipynb>`_
  Core data structures for working with unstructured grid and data files
 
-`Plotting API <user-guide/plotting.ipynb>`_
- Visualize unstructured grid datasets using UXarray's plotting api
+`Loading Data with Dask <user-guide/parallel-load-ux-with-dask.ipynb>`_
```
Comment (philipc2) [inline docs/userguide.rst:34]:
```suggestion
`Loading Data using Dask <user-guide/parallel-load-ux-with-dask.ipynb>`_
```

File: docs/gallery.yml [hunk 3]
```diff
@@ -7,11 +7,18 @@ visualization-examples:
     path: examples/visualization/mpas_topology.ipynb
     thumbnail: _static/thumbnails/mpas_topology.png
 
+  - title: Plotting E3SMv2 Output with UXarray
```
Comment (philipc2) [inline docs/gallery.yml:10]: E3SMv2 Output
```suggestion
  - title: E3SMv2 Model Output Visualizations 
```

File: docs/gallery.yml [hunk 4]
```diff
@@ -7,11 +7,18 @@ visualization-examples:
...
+    path: examples/visualization/e3sm-load-viz.ipynb
+    thumbnail: _static/thumbnails/ts_e3sm_load.png
+
 workflow-examples:
   - title: Working with MPAS Grids
     path: examples/working-with-mpas-grids.ipynb
     thumbnail: _static/thumbnails/004-thumbnail.png
 
+  - title:  Calculations on E3SMv2 Output with UXarray
```
Comment (philipc2) [inline docs/gallery.yml:19]: An idea for a more descriptive title, any thoughts? 

```suggestion
  - title:  Radiative Feedback Analysis on E3SMv2 Model Output
```

**Review by philipc2 — CHANGES_REQUESTED** [review]
Apologies for the long gap in reviews! I've left a few suggestions here. Once we've resolved all the ReviewNB comments, we should be good to go!

**Comment by rytam2** [comment]
<p>Now updated with link.</p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2253217159'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by rytam2** [comment]
<p>Updated</p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2253217174'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by rytam2** [comment]
<p>Related to bug in <code>.plot.rasterize</code> that requires a name to plot; will keep this line as a temporary solution.  </p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2252958987'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by philipc2 — CHANGES_REQUESTED** [review]
Excellent work Rachel! I've put a few comments in ReviewNB.

May we change the thumbnail to one of the projected ones?

image


I'd also consider a thumbnail for the e3sm calc notebook.

**Comment by rytam2** [comment]
<p>accepted</p>

---

 <a href='https://app.reviewnb.com/UXARRAY/uxarray/pull/836/discussion/#comment-2400949092'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by philipc2 — APPROVED** [review]
Great work Rachel!
